From c90388ee1c62f130cb7fcd9984b9e2d897d88e74 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sun, 4 Jan 2026 00:21:37 -0500
Subject: [PATCH 472/479] Revert "HACK: mfplat: Use the MP4 bytestream handler
 as a fallback."

This reverts commit dbf857a2d56e6ecca524bb96efc00f2893f28253.
---
 dlls/mfplat/main.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/dlls/mfplat/main.c b/dlls/mfplat/main.c
index 0a7c294294e..1a24d380d0d 100644
--- a/dlls/mfplat/main.c
+++ b/dlls/mfplat/main.c
@@ -43,7 +43,6 @@
 #include "strsafe.h"
 #undef INITGUID
 #include "evr.h"
-#include "wine/mfinternal.h"
 /* mfd3d12 guids are not included in mfuuid */
 #define INITGUID
 #undef EXTERN_GUID
@@ -6299,9 +6298,10 @@ static HRESULT resolver_get_bytestream_url_hint(IMFByteStream *stream, WCHAR con
     return S_OK;
 }
 
-static HRESULT resolver_create_default_handler(IMFByteStreamHandler **handler)
+static HRESULT resolver_create_gstreamer_handler(IMFByteStreamHandler **handler)
 {
-    return CoCreateInstance(&CLSID_MPEG4ByteStreamHandlerPlugin, NULL, CLSCTX_INPROC_SERVER, &IID_IMFByteStreamHandler, (void **)handler);
+    static const GUID CLSID_GStreamerByteStreamHandler = {0x317df618, 0x5e5a, 0x468a, {0x9f, 0x15, 0xd8, 0x27, 0xa9, 0xa0, 0x81, 0x62}};
+    return CoCreateInstance(&CLSID_GStreamerByteStreamHandler, NULL, CLSCTX_INPROC_SERVER, &IID_IMFByteStreamHandler, (void **)handler);
 }
 
 static HRESULT resolver_get_bytestream_handler(IMFByteStream *stream, const WCHAR *url, DWORD flags,
@@ -6337,14 +6337,12 @@ static HRESULT resolver_get_bytestream_handler(IMFByteStream *stream, const WCHA
        this handler for all possible types.
      */
 
-    TRACE( "url_ext %s mimeW %s\n", debugstr_w(url_ext), debugstr_w(mimeW) );
-
     if (url_ext || mimeW)
     {
         hr = resolver_create_bytestream_handler(stream, flags, mimeW, url_ext, handler);
 
         if (FAILED(hr))
-            hr = resolver_create_default_handler(handler);
+            hr = resolver_create_gstreamer_handler(handler);
     }
 
     CoTaskMemFree(mimeW);
@@ -6362,7 +6360,7 @@ static HRESULT resolver_get_bytestream_handler(IMFByteStream *stream, const WCHA
     hr = resolver_create_bytestream_handler(stream, flags, NULL, url_ext, handler);
 
     if (FAILED(hr))
-        hr = resolver_create_default_handler(handler);
+        hr = resolver_create_gstreamer_handler(handler);
 
     return hr;
 }
-- 
2.52.0

